packages/shared/src/components/post/SharePostContent.tsx

```diff
@@ -39,82 +41,98 @@ const SharePostContentSkeleton = () => (
   </>
 );
 
-function SharePostContent({
-  post,
+export function CommonSharePostContent({
+  sharedPost,
+  mainSource,
   onReadArticle,
-}: SharePostContentProps): ReactElement {
+}: CommonSharePostContentProps): ReactElement {
   const { openNewTab } = useContext(SettingsContext);
   const openArticle = (e: React.MouseEvent) => {
     e.stopPropagation();
     onReadArticle();
   };
 
-  if (!post.sharedPost) {
+  if (!sharedPost) {
     return <SharePostContentSkeleton />;
   }
 
   const shouldUseInternalLink =
-    isSharedPostSquadPost(post) || isInternalReadType(post.sharedPost);
+    isSharedPostSquadPost({ sharedPost }) || isInternalReadType(sharedPost);
```

Comment:
This was changed because SharePostContent is now used both on real posts and pending ones (in preview modal). 
Everything is marked as changes, but seems that the component is the same, just split between the common part and the full-feature one. 

<img width="300" alt="image" src="https://github.com/user-attachments/assets/2ab70b0b-21ce-48dc-9024-4642bfccf9dd">